docs/database-connection-string.md

```diff
@@ -1,85 +1,83 @@
-* `Never` - Leaves database processes running, even after the application shuts down and even if the application started them.
+### ContextCount
 
-_Note: the `Never` option is not effective when the application process is forcibly killed with task manager or `Ctrl + C`. In this case operating system will kill corresponding Starcounter processes as well._
+The ContextCount option defines the number of **database contexts** to use in the database connection.
 
-### The `ContextCount` option
+"Database context", as used here, should not be confused with `IDatabaseContext` as used in the .NET API for transactions. In the context of database connections, database contexts refer to the entrypoints to the database from the applications' point of view. When an application thread is assigned a database context, it can read and write to the database.
 
-Starcounter has maximum of `31` database contexts available for allocation between multiple processes. By default, Starcounter application tries to allocate double as many contexts as CPU cores available, but minimum `2` and maximum `24`.
+Each Starcounter database has a maximum of 31 database contexts available for allocation between multiple application processes. By default, each Starcounter application tries to allocate twice as many contexts as CPU cores available, with a minimum of 2 and a maximum of 24.
 
-The ideal configuration is to have a database context per CPU core. Having more database contexts than CPU cores might increase performance in certain concurrent database access workloads. Having less database contexts than CPU cores might decrease performance of concurrent database access during high concurrent load.
+The ideal configuration is to have a database context per CPU core. Having more database contexts than CPU cores might increase performance in certain concurrent database access workloads. Having fewer database contexts than CPU cores might decrease performance of concurrent database access during high concurrent load.
 
-Use `ContextCount` option to manually adjust database contexts allocation.
+Use `ContextCount` option to manually adjust the number of database contexts to occupy in this connection.
 
-## Connection String sample
+## Example
 
 ```text
 Database=./.database/Sample;OpenMode=CreateIfNotExists;StartMode=StartIfNotRunning;StopMode=IfWeStarted;ContextCount=10
 ```
 
+This string defines the database option using a relative path to a database `Sample`, located in the `.database` directory of the current directory. If such a directory does not exist at the time of connection, the OpenMode option of `CreateIfNotExists` defines that it should be created. If not running already, the StartMode option of `StartIfNotRunning` defines that it should be started. When te app disconnects from the database, it will stop the database process if it started, due to the StopMode option being defined as `IfWeStarted`. Lastly, we set the number of database contexts to 10.
```

Comment:
```suggestion
This string defines the database option using a relative path to a database `Sample`, located in the `.database` folder in the current directory. If such a directory does not exist at the time of connection, the OpenMode option of `CreateIfNotExists` defines that it should be created. If not running already, the StartMode option of `StartIfNotRunning` defines that it should be started. When te app disconnects from the database, it will stop the database process if it started, due to the StopMode option being defined as `IfWeStarted`. Lastly, we set the number of database contexts to 10.
```